lalrpop-util/src/lib.rs

```diff
@@ -48,11 +49,14 @@ pub mod state_machine;
 pub enum ParseError<L, T, E> {
     /// Generated by the parser when it encounters a token (or EOF) it did not
```

Comment:
Just noticed while looking at #992.  This case is generated by the *lexer*, not the parser (which is the difference between this and UnrecognizedToken).  Let's get this and the UnrecognizedToken cleaned up as part of this PR.  I'll push a fix in a bit.

Reply:
This sounds related to https://github.com/lalrpop/lalrpop/issues/707. Maybe the built-in lexer errors should be better decoupled from the parser such that users and lalrpop can leverage the same lexer error interface.

Reply:
Yeah, maybe?  I think my main thought on #707 is that I'm not really sure how ParseError::user doesn't get the job done?  But a LexerError variant would probably make sense.  I guess the question is what is the type of that?  A new generic in ParseError (as mentioned in #707)?  That's already a complicated signature (which has an error generic).  I'm also not clear on how that approach works for users of the built in lexer?

If instead we define LexerError as an enum ourselves, then we have to have a custom field in there, since we can't enumerate all user error types, so it's extra indirection.

I'm definitely in favor of improvements to ParseError, but neither of those paths sound initially promising to me...

I also wondered about extending lexer error coverage for #992 to let lexers return recoverable and non-recoverable errors, but I don't really see the benefit of a lexer returning `Err(RecoverableError(_))` rather than `Ok(Token::Error)` like in the workaround code.